packages/snap/src/transaction.ts

```diff
@@ -23,9 +41,9 @@ export async function send(node, transaction, sendingMode, publicKey) {
     encodeInputData(transaction, latestBlock),
   ]);
 
-  const keyPair = await deriveKeyPair(0);
+  const keyPair = await findKeyPair(publicKey);
 
-  if (keyPair.publicKey.toString() !== publicKey) {
+  if (keyPair === null) {
     throw invalidParameters('Uknown public key');
```

Comment:
Typo in error message

Reply:
oh right, it was there already tho. nice catch